mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogEventPool.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Log {
+
+	sealed class LogEventPool {
```

Comment:
One thing we can do is have a `Clone` method on all these `LogEvent` classes so that a user can easily make a copy if they really want to keep a reference around.